### PR TITLE
feat(tf): support tensor fitting with hybrid descriptor

### LINKED ISSUE #4527: [BUG] Dipole training not working with any hybrid descriptor combination
### Bug summary

File "/home/analys/.local/lib/python3.8/site-packages/deepmd/fit/dipole.py", line 88, in __init__
    self.dim_rot_mat_1 = descrpt.get_dim_rot_mat_1()
  File "/home/analys/.local/lib/python3.8/site-packages/deepmd/descriptor/descriptor.py", line 140, in get_dim_rot_mat_1
    raise NotImplementedError
NotImplementedError

### DeePMD-kit Version

DeePMD-kit v2.2.11

### Backend and its version

TensorFlow 2.13.0

### How did you download the software?

Offline packages

### Input Files, Running Commands, Error Log, etc.

{
  "",

  "model": {
    "type_map": [
      "C",
      "N",
      "O",
      "H"
    ],

    "descriptor": {
      "type": "hybrid",
      "list": [
        {
          "type": "se_e3",
          "sel": "auto",
          "rcut_smth": 0.5,
          "rcut": 3.0,
          "neuron": [16, 32, 64],
          "resnet_dt": false,
          "seed": 31,
          "_comment": "Short-range descriptor: se_e3"
        },
        {
          "type": "se_e2_a",
          "sel": "auto",
          "rcut_smth": 0.5,
          "rcut": 10.0,
          "neuron": [111, 111, 111],
          "axis_neuron": 16,
          "resnet_dt": false,
          "seed": 31,
          "_comment": "comment"
        }
      ],
      "_comment": "Double-hybrid descriptor (se_e3 + se_e2_a)"
    },

    "fitting_net": {
      "type": "dipole",
      "neuron": [
        512,
        512,
        512
      ],
      "resnet_dt": true,
      "seed": 31,
      "activation_function": "relu",
      "_comment": "Dipole fitting network"
    },

    "_comment": "End of model block"
  },

  "learning_rate": {
    "type": "exp",
    "decay_steps": 1000,
    "start_lr": 1e-3,
    "stop_lr": 1e-8,
    "_comment": "Exponential decay learning rate schedule"
  },

  "loss": {
    "type": "tensor",
    "pref": 1.0,
    "pref_atomic": 0.0,
    "_comment": "Loss for dipole training (using 'tensor')"
  },

  "training": {
    "training_data": {
      "systems": [
        "./training_data"
      ],
      "batch_size": "auto",
      "_comment": "Specify your training set"
    },
    "validation_data": {
      "systems": [
        "./validation_data"
      ],
      "batch_size": "auto",
      "_comment": "Specify your validation set"
    },
    "numb_steps": 80000,
    "seed": 555,
    "disp_file": "lcurve.out",
    "disp_freq": 200,
    "save_freq": 500,
    "_comment": "Overall training parameters"
  },

  "_comment": "That's all"
}

### Steps to Reproduce

Try fitting dipoles with hybrid

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by njzjz**
Fix #4527.

## Summary by CodeRabbit

- **New Features**
	- Added rotation matrix handling methods to descriptor classes.
	- Introduced validation for model initialization.

- **Bug Fixes**
	- Improved error handling for fitting network configuration.

- **Tests**
	- Added new test suite for dipole hybrid descriptor model validation.

- **Refactor**
	- Enhanced descriptor class functionality with default rotation matrix methods.

### CODE REVIEW

File: source/tests/tf/test_dipole_hybrid_descrpt.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        t_fparam = None
+
+        model_pred = model.build(
+            t_coord,
+            t_type,
+            t_natoms,
+            t_box,
+            t_mesh,
+            t_fparam,
+            suffix="dipole_se_a",
```
Comment (njzjz) [inline source/tests/tf/test_dipole_hybrid_descrpt.py:122]:
```suggestion
            suffix="dipole_hybrid",
```

File: source/tests/tf/test_dipole_hybrid_descrpt.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    def setUp(self) -> None:
+        gen_data()
+
+    def test_model(self) -> None:
+        jfile = "polar_se_a.json"
+        jdata = j_loader(jfile)
+
+        systems = jdata["systems"]
+        set_pfx = "set"
+        batch_size = jdata["batch_size"]
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/tests/tf/test_dipole_hybrid_descrpt.py:38]:
```suggestion
```

File: source/tests/tf/test_dipole_hybrid_descrpt.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        test_size = jdata["numb_test"]
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/tests/tf/test_dipole_hybrid_descrpt.py:39]:
```suggestion
```

**Review by wanghan-iapcm — APPROVED** [review]
